sycl/include/sycl/detail/generic_type_traits.hpp

```diff
@@ -214,14 +213,23 @@ inline constexpr bool is_ugenlonginteger_v =
 template <typename T>
 inline constexpr bool is_geninteger_v = is_contained_v<T, gtl::integer_list>;
 
+template <typename T>
+using is_geninteger = std::bool_constant<is_geninteger_v<T>>;
+
```

Comment:
Why do we need these?

Reply:
`is_geninteger*bit_v` passes the templated type onto `is_gen_based_on_type_sizeof`.